### ISSUE TITLE
[QUESTION]: Why is it that when I train protacs large molecules with chemprop the results fluctuate a lot, but this is not the case when training small molecules

### ISSUE THREAD

**Issue by 315954193** [chemprop/chemprop#545]
**What are you trying to do?**
I want to use the ChemProp regression model to train and predict the retention times of PROTACs

**Previous [attempts**]
_**parameter  setting：**_
small molecular 
{'activation': 'ReLU',                                                               
 'atom_messages': False,
 'batch_size': 50,
 'bias': False,
 'checkpoint_dir': None,
 'checkpoint_path': None,
 'checkpoint_paths': None,
 'config_path': None,
 'crossval_index_dir': None,
 'crossval_index_file': None,
 'cuda': True,
 'data_path': 'RT_R1.csv',
 'dataset_type': 'regression',
 'depth': 3,
 'dropout': 0.0,
 'ensemble_size': 1,
 'epochs': 50,
 'features_generator': None,
 'features_only': False,
 'features_path': None,
 'features_scaling': True,
 'ffn_hidden_size': 300,
 'ffn_num_layers': 2,
 'final_lr': 0.0001,
 'folds_file': None,
 'gpu': 0,
 'hidden_size': 300,
 'init_lr': 0.0001,
 'log_frequency': 10,
 'max_data_size': None,
 'max_lr': 0.001,
 'metric': 'rmse',
 'minimize_score': True,
 'multiclass_num_classes': 3,
 'no_cache': False,
 'num_folds': 5,
 'num_lrs': 1,
 'quiet': False,
 'save_dir': 'RT_R1_rand/fold_0',
 'save_smiles_splits': True,
 'seed': 0,
 'separate_test_features_path': None,
 'separate_test_path': None,
 'separate_val_features_path': None,
 'separate_val_path': None,
 'show_individual_scores': False,
 'split_sizes': [0.8, 0.1, 0.1],
 'split_type': 'random',
 'test': False,
 'test_fold_index': None,
 'undirected': False,
 'use_compound_names': False,
 'use_input_features': None,
 'val_fold_index': None,
 'warmup_epochs': 2.0}
large molecular
{'activation': 'ReLU',
 'atom_messages': False,
 'batch_size': 50,
 'bias': False,
 'checkpoint_dir': None,
 'checkpoint_path': None,
 'checkpoint_paths': None,
 'config_path': None,
 'crossval_index_dir': None,
 'crossval_index_file': None,
 'cuda': True,
 'data_path': '1106rt.csv',
 'dataset_type': 'regression',
 'depth': 3,
 'dropout': 0.0,
 'ensemble_size': 1,
 'epochs': 50,
 'features_generator': None,
 'features_only': False,
 'features_path': None,
 'features_scaling': True,
 'ffn_hidden_size': 300,
 'ffn_num_layers': 2,
 'final_lr': 0.0001,
 'folds_file': None,
 'gpu': 0,
 'hidden_size': 300,
 'init_lr': 0.0001,
 'log_frequency': 10,
 'max_data_size': None,
 'max_lr': 0.001,
 'metric': 'rmse',
 'minimize_score': True,
 'multiclass_num_classes': 3,
 'no_cache': False,
 'num_folds': 5,
 'num_lrs': 1,
 'quiet': False,
 'save_dir': '1106rt_check/fold_0',
 'save_smiles_splits': True,
 'seed': 0,
 'separate_test_features_path': None,
 'separate_test_path': None,
 'separate_val_features_path': None,
 'separate_val_path': None,
 'show_individual_scores': False,
 'split_sizes': [0.8, 0.1, 0.1],
 'split_type': 'random',
 'test': False,
 'test_fold_index': None,
 'undirected': False,
 'use_compound_names': False,
 'use_input_features': None,
 'val_fold_index': None,
 'warmup_epochs': 2.0}
Code command line：
chemp1_train --gpu 0 --data_path xxx.csv --dataset_type regression --save_dir xxx_rand_check --epochs 50 --split_type random \
--save_smiles_splits --num_folds 5 --features_path  xxx.csv

 
**Screenshots**
The results are listed in the table, there is a big difference between the rmse obtained from the large molecule's and the small molecule's regression training with the chemprop model in the same environment, and the fluctuation of the rmse of the large molecule's is more obvious, I'd like to know the reason for this?
[reg_result.csv](https://github.com/chemprop/chemprop/files/13645278/reg_result.csv)
The dataset files for large and small molecules are as follows：
[large_mol.csv](https://github.com/chemprop/chemprop/files/13645335/large_mol.csv)
[small_mol.csv](https://github.com/chemprop/chemprop/files/13645336/small_mol.csv)

**Comment by kevingreenman** [comment]
I don't know anything about PROTACs so won't be able to give any domain-specific insights, but the first thing that jumps out to me is that `small_mols.csv` has almost 70x the data as `large_mols.csv`, so it's not surprising that the much smaller dataset performs much worse. Larger fluctuations in cross-validation are also to be expected for a smaller dataset.

It's interesting that the results for the larger molecules become so much worse when you add additional fingerprint features, whereas that doesn't happen for the smaller molecules. I don't have a great explanation of why that might be.

A few things to consider/try:
* Training a single model on both the large and small molecules together but still calculating the RMSE of the large and small predictions separately. It's possible this will give a similar result to what you have now since the large molecules would be very underrepresented in the training set, but it's also possible that the model might get better at predicting the large molecules based on what it's learned from the more numerous small molecules.
* If the above doesn't help due to the unbalanced dataset sizes, you can try [weighting](https://github.com/chemprop/chemprop?tab=readme-ov-file#aggregation) the datapoints corresponding to the large molecules more heavily during the training process.
* Changing the [aggregation function argument](https://github.com/chemprop/chemprop?tab=readme-ov-file#aggregation) - if your property of interest is extensive (depends on the size of the molecule), sum and norm aggregations are expected to perform better than the default mean.

**Comment by kevingreenman** [comment]
Closing this for now. Feel free to reopen if you have additional questions.
